sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/csv/CSVParameters.scala

```diff
@@ -44,9 +44,11 @@ private[sql] case class CSVParameters(parameters: Map[String, String]) extends L
     }
   }
 
-  val delimiter = CSVTypeCast.toChar(parameters.getOrElse("delimiter", ","))
+  val delimiter = CSVTypeCast.toChar(
+    parameters.getOrElse("delimiter", parameters.getOrElse("sep", ",")))
```

Comment:
while you are at it, can you make parameters map transient, and make it not a case class?